city_scrapers/spiders/il_police_professionalism.py

```diff
@@ -0,0 +1,89 @@
+from time import strptime
+from datetime import datetime
+
+from city_scrapers_core.constants import (COMMISSION)
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class IlPoliceProfessionalismSpider(CityScrapersSpider):
+    name = "il_police_professionalism"
+    agency = "Illinois Commission on Police Professionalism"
+    timezone = "America/Chicago"
+    allowed_domains = ["www.isp.state.il.us"]
+    start_urls = ["https://www.isp.state.il.us/media/openmtgs.cfm"]
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        for item in response.xpath("//*[contains(text(),\"Police Professionalism\")]"):
```

Comment:
Sometimes agencies will change their markup a bit so that the actual text might be nested inside of a `strong` tag inside the `a` tag, so the selector `"li[style] a"` might be safer to just find links with the `style` attribute on them which only seems to match document links that we're interested in. Then you can get the text of each item with:

```python
item_text = " ".join(item.css("*::text").extract())
```

This way, you pull all of the text within the item, even if it's nested. From there you can check if "Police Professionalism" is included

Reply:
I wrote the following code which basically does the same thing

```
for item in response.xpath("//li[contains(@style,'padding-bottom')]"):
            link = item.xpath('a[(.//*|.)[contains(text(),"Police Professionalism")]]/@href'
                              ).extract()
```